src/apps/bridge/reportBuilder.cpp

```diff
@@ -0,0 +1,490 @@
+                    APP_PUB_SUB_BM_BRIDGE_SENSOR_REPORT_TYPE,
+                    APP_PUB_SUB_BM_BRIDGE_SENSOR_REPORT_VERSION);
+                vPortFree(message_buff);
+              } else {
+                printf("No nodes to send data for\n");
+              }
+              printf("Clearing the list\n");
+              _ctx._reportBuilderLinkedList.clear();
+            }
+            printf("Clearing the sample counter\n");
+            _ctx._sample_counter = 0;
+            // Clear the report periods network associated data to rebuild it for the next report period
+            _ctx._report_period_num_nodes = 0;
+            _ctx._report_period_max_network_crc32 = 0;
+            memset(_ctx._report_period_node_list, 0, sizeof(_ctx._report_period_node_list));
+          }
+          break;
+        }
+
+        case REPORT_BUILDER_SAMPLE_MESSAGE: {
+          if (_ctx._samplesPerReport > 0) {
+            printf("Adding sample for %" PRIx64 " to list\n", item.node_id);
+            _ctx._reportBuilderLinkedList.findElementAndAddSampleToElement(item.node_id, item.sensor_type, item.sensor_data, item.sensor_data_size, _ctx._samplesPerReport, _ctx._sample_counter);
+          } else {
+            printf("samplesPerReport is 0, not adding sample to list\n");
+          }
+          break;
+        }
+
+        case REPORT_BUILDER_CHECK_CRC: {
+
+          // IMPORTANT: If the bus is constantly on, the currentAggPeriodMin(or sample period eventually) must be longer than the topology sampler period (1 min)
+          // otherwise the CRC for the report builder may not update before a message needs to be sent
+
+          printf("Checking CRC in report builder!\n");
+          // Get the latest crc - if it doens't match our saved one, pull the latest topology and compare the topology
+          // to our current topology - if it doesn't match our current one (and mainly if it is bigger or the same size) then
+          // update the crc and the topology list
+          uint32_t temp_network_crc32 = 0;
+          uint32_t temp_cbor_config_size = 0;
+          uint8_t *last_network_config = topology_sampler_alloc_last_network_config(temp_network_crc32, temp_cbor_config_size);
+          if (last_network_config != NULL) {
+            printf("Got CRC %" PRIx32 " OLD CRC %" PRIx32 "\n", temp_network_crc32, _ctx._report_period_max_network_crc32);
+            if (temp_network_crc32 != _ctx._report_period_max_network_crc32) {
+              printf("Getting topology in report builder!\n");
+              uint64_t temp_node_list[TOPOLOGY_SAMPLER_MAX_NODE_LIST_SIZE];
+              size_t temp_node_list_size = sizeof(temp_node_list);
+              uint32_t temp_num_nodes;
+              if (topology_sampler_get_node_list(temp_node_list, temp_node_list_size, temp_num_nodes, TOPO_TIMEOUT_MS)) {
+                // TODO - we need to use the cbor_map and build a report builder topology map that only
```

Comment:
Why do we need an aanderaa specific report builder topology?

Reply:
The behavior rn is that if there is a node we will add it to the report, even if it is not an aanderaa node. This means that a unit with a devkit will be sending extra data. I asked Evan and he said there will be some DVT2 customers using additional devkit nodes, or for our own testing the loadcell. 

I suppose we don't need an aanderaa specific topology, but more a map that associates a node_id in the topo list with it being an aanderaa, that way when the report builder loops throught the topology, we know when to look for/add a sensors samples to a report.

Reply:
Ah I see, so we need something like a "supported schema type" or something where we can first add aanderaa and then eventually the RBR and soft?

Reply:
Thats one aspect. I think we also need report builder to save a dict with "node_id : isSupportedSensorSchema, ...." such that if a sensor did not generate any samples during a report period, but is a part of the "max topology" the report builder will know if it can add NAN's to the report, or if it should not add anything.